powerscale/provider/s3_bucket_resource.go

```diff
@@ -0,0 +1,459 @@
+func NewS3BucketResource() resource.Resource {
+	return &S3BucketResource{}
+}
+
+// S3BucketResource defines the resource implementation.
+type S3BucketResource struct {
+	client *client.Client
+}
+
+// Configure configures the resource.
+func (r *S3BucketResource) Configure(ctx context.Context, req resource.ConfigureRequest, res *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	c, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		res.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *c.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.client = c
+}
+
+// Metadata describes the resource arguments.
+func (r *S3BucketResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_s3_bucket"
+}
+
+// Schema describes the resource arguments.
+func (r *S3BucketResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "This resource is used to manage the S3 Bucket entity of PowerScale Array. PowerScale S3 Bucket map to the PowerScale file system as base directory for Objects. We can Create, Update and Delete the S3 Bucket using this resource. We can also import an existing S3 Bucket from PowerScale array.",
+		Description:         "This resource is used to manage the S3 Bucket entity of PowerScale Array. PowerScale S3 Bucket map to the PowerScale file system as base directory for Objects. We can Create, Update and Delete the S3 Bucket using this resource. We can also import an existing S3 Bucket from PowerScale array.",
+		Attributes: map[string]schema.Attribute{
+			"acl": schema.ListNestedAttribute{
+				Description:         "Specifies properties for an S3 Access Control Entry.",
+				MarkdownDescription: "Specifies properties for an S3 Access Control Entry.",
+				Computed:            true,
+				Optional:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"grantee": schema.SingleNestedAttribute{
+							Description:         "Specifies the persona of the file group.",
+							MarkdownDescription: "Specifies the persona of the file group.",
+							Required:            true,
+							Attributes: map[string]schema.Attribute{
+								"id": schema.StringAttribute{
+									Description:         "Specifies the serialized form of a persona, which can be 'UID:0', 'USER:name', 'GID:0', 'GROUP:wheel', or 'SID:S-1-1'.",
+									MarkdownDescription: "Specifies the serialized form of a persona, which can be 'UID:0', 'USER:name', 'GID:0', 'GROUP:wheel', or 'SID:S-1-1'.",
+									Computed:            true,
+									Optional:            true,
+								},
+								"name": schema.StringAttribute{
```

Comment:
Like https://github.com/dell/terraform-provider-powerscale/pull/144, `id` and `name+type` may be mismatched. Please check the REST API behavior for the mismatch. Probably it would be a subject we should discuss.

Reply:
It won't return `id` in the http response. And current behavior of setting acl when creating s3 bucket from powerscale web portal is also just to pass `name+type`. The following is the json payload captured from web browser FYI.
```
{"name":"test","path":"/ifs/home","owner":"root","acl":[{"grantee":{"name":"Everyone","type":"wellknown"},"permission":"READ"}]}
```
Having `id` here is just to align with the API definition in `PowerScale_API_9.5.0.json`.

Reply:
If we pass `id` in the API call, it will simply ignore the value of `name + type`. And inconsistency error will report if we pass wrong `id` & `name + type` combination.

Reply:
> If we pass `id` in the API call, it will simply ignore the value of `name + type`. And inconsistency error will report if we pass wrong `id` & `name + type` combination.

Our code should be consistent. Any thoughts @taohe1012 , @RayLiu7 , @forrestxia , @P-Cao , @NavisJ ?

Reply:
Hi @vangork , since `id` is not accepted by REST API, how about setting it to `Computed` only, thus user will not input an unexpected value? and the description should be revised a bit, for example, remove `Specifies` because `id` is read-only.

Reply:
@baoy1 Since creating/updating acl by `id` and `type+name` are both supported, and the limited test http response would only return `type+name`, I would keep the current implementation of setting id, name, type to be all optional + computed to be compatible with the swagger definition of `v1AuthAccessAccessItemFileGroup`. Meanwhile document at the terraform side: "Please input type+name only unless id is must when setting acl". Dose it make sense?

Reply:
Document "type+name is preferable" may be an expedient solution. It would be a bit confusing if user inputs id as `SID:S-1-1-0`, but after terraform apply, id is `null`, though it is REST API behavior.

Reply:
As discussed in the scrum meeting, set `id` to be `Computed` only.

Reply:
@vangork, are `type`+`name` mandatory?

Reply:
yes.

Reply:
Nice catch! Changed to `Required: true` for `type+name`